main.tf

```diff
@@ -1,152 +1,165 @@
-  pub_lb_subnet_id = lookup(module.network.subnet_ids, "pub_lb")
-
-  # workers ip addresses for backend resources
-  olcne_workers = module.worker.worker_ip_list
+module "vault" {
+  source     = "./modules/terraform-oci-vault"
+  count      = var.use_vault ? 1 : 0
+
+  user_id                  = var.user_id
+  region                   = var.region
+  tenancy_id               = var.tenancy_id
+  fingerprint              = var.fingerprint
+  api_private_key_path     = var.api_private_key_path
+  private_key_password     = var.private_key_password
+  availability_domain_id   = var.availability_domain_id
+  compartment_id           = var.compartment_id
+  prefix                   = "${var.prefix}-vault"
+  subnet_id                = var.deploy_networking ? module.oci-ocne-network[0].private_subnet_id.*.id[0] : var.subnet_id
+  instance_shape           = var.instance_shape
+  ssh_public_key_path      = var.ssh_public_key_path
+  ssh_private_key_path     = var.ssh_private_key_path
+  load_balancer_port       = "8200"
+  proxy                    = var.proxy
+  vault_ocid               = var.vault_ocid
+  key_ocid                 = var.key_ocid
+  vault_namespace          = var.vault_namespace
+  load_balancer_ocid       = module.infrastructure.load_balancer_ocid
+  create_load_balancer     = false
+  load_balancer_ip         = module.infrastructure.load_balancer_ip
+  pool_size                = var.vault_pool_size
+  secret_name              = local.secret_name
+  ocne_secret_name        = local.ocne_secret_name
+  load_balancer_shape      = var.load_balancer_shape
+  enable_bastion          = var.enable_bastion
+  bastion_public_ip        = var.enable_bastion ? module.oci-ocne-network[0].bastion_public_ip : var.bastion_public_ip
+  bastion_user             = var.bastion_user
+  bastion_private_key_path = var.enable_bastion || var.bastion_public_ip != "" ? var.bastion_private_key_path : ""
+  compute_user             = var.compute_user
+  freeform_tags            = var.freeform_tags
 }
 
-# configuration
-module "olcne" {
-  source = "./modules/olcne"
-
-  # general oci parameters
-  compartment_id = var.compartment_id
-  label_prefix   = var.label_prefix
-  region         = var.region
-
-  # bastion
-  bastion_public_ip    = module.base.bastion_public_ip
-  ssh_private_key_path = var.ssh_private_key_path
-
-  # operator
-  operator_ip = module.base.operator_private_ip
-
-  # list of master nodes
-  olcne_masters = local.olcne_masters
-
-  # private CA Certificate
-  olcne_certificate = local.olcne_certificate
-
-  # list of worker nodes
-  olcne_workers = local.olcne_workers
-
-  # private key parameter to access other nodes
-  secret_id = var.secret_id
-
-  # olcne environment
-  olcne_environment = local.olcne_environment
-
-  helm_version = var.helm_version
-
-  loadbalancer_ip_address = module.loadbalancer.pub_lb_ip
-  deploy_nginx_software = var.deploy_nginx_software
+module "ocne-provision" {
```

Comment:
as above, can you please make a short comment about what each module does so we can follow?